### ISSUE TITLE
ContaoModuleBundle not compatible with Symfony 4 flex

### ISSUE THREAD

**Issue by christian-kolb** [contao/contao#216]
**Affected version(s)**
4.*

**Description**  
When running Contao as a Symfony bundle, you have to include the "old modules" to your `AppKernel` through the `ContaoModuleBundle` like for example:

```
public function registerBundles(): array
{
    $bundles = [
        ...
        new Contao\CoreBundle\HttpKernel\Bundle\ContaoModuleBundle('multicolumnwizard', $this->getRootDir()),
        ...
}
```

Which means you have to use parameters for the initialization of the module bundle. 

In Symfony 4 there is a `bundles.php`, where the bundles are listed the following way:

```
return [
    Symfony\Bundle\FrameworkBundle\FrameworkBundle::class => ['all' => true],
    ...
```

Meaning there is no way to provide any parameter for the module initialization.
Is there some other way to enable loading Contao modules in Symfony 4?

**How to reproduce**  

[Create a new Symfony project with Symfony flex](https://symfony.com/doc/current/setup.html) and try to add a Contao module with `ContaoModuleBundle`.

**Comment by christian-kolb** [comment]
Am I the only one using Contao as a bundle? Does anyone have a solution for the issue?

**Comment by ausi** [comment]
According to https://github.com/symfony/symfony/issues/25913#issuecomment-360504168 this is not related to Symfony 4 but to Symfony Flex:

> Note that Symfony 4 does not forbid passing constructor arguments to bundles. This is Flex not supporting it. Someone using SF 4 without Flex is not affected by this, and someone using Flex with Symfony 3.4 has this issue.

I think you have to register the bundles manually in the `Kernel` of your project as described in https://github.com/symfony/symfony/issues/25913#issuecomment-360505078

**Comment by christian-kolb** [comment]
@ausi Thanks a lot for the information. As it won't be fixed in flex it really doesn't make sense for Contao to "fix" something here. Then I will try to initiate those bundles directly.

**Comment by ausi** [comment]
Maybe it’s possible to reuse the `ContaoKernel::registerBundles()` method for loading the Contao specific bundles?

**Comment by christian-kolb** [comment]
You mean a Contao specific way of loading them without the need to adapt the Kernel?

**Comment by ausi** [comment]
No, I think you have to adapt the Kernel of your project, but maybe you could call `ContaoKernel::registerBundles()` from your Kernel so that you don’t have to manually register all Contao modules?

**Comment by christian-kolb** [comment]
Ah ok, I will check and reply as soon as I figured it out : )

**Comment by fritzmg** [comment]
@christian-kolb have you ever tested it? :)

**Comment by christian-kolb** [comment]
@fritzmg I don't think so, otherwise I'm sure I would have commented it here. I think back when I tried I ran into multiple other issues with Symfony and Contao. Including but not limited to how Doctrine and DCA didn't play nice with each other and this was the point where I abandoned this combination. I haven't tried since as I haven't had a usecase where I needed Contao that badly.
Sorry that I'm no help there
